Lib/typing.py

```diff
@@ -1939,12 +1941,9 @@ def __new__(mcls, name, bases, namespace, /, **kwargs):
                         f"Protocols can only inherit from other protocols, "
                         f"got {base!r}"
                     )
-        return super().__new__(mcls, name, bases, namespace, **kwargs)
-
-    def __init__(cls, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        if getattr(cls, "_is_protocol", False):
-            cls.__protocol_attrs__ = _get_protocol_attrs(cls)
+        cls = super().__new__(mcls, name, bases, namespace, **kwargs)
+        cls.__protocol_attrs_cache__ = None
```

Comment:
we _could_ avoid setting this for nomimal subclasses of protocols classes (we never access the `__protocol_attrs__` or `__non_callable_proto_members__` properties for nominal subclasses, and end users definitely shouldn't be either, so exactly what you get when you access those properties on nominal subclasses is a bit of a moot point)

```suggestion
        if getattr(cls, "_is_protocol", False):
            cls.__protocol_attrs_cache__ = None
```

not sure how much memory that would actually save, though -- probably not much

Reply:
Doing this in a slightly different way, we can avoid the getattr().